cmd/addvalidators.go

```diff
@@ -0,0 +1,79 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package cmd
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/z"
+)
+
+type addValidatorsConfig struct {
+	Lockfile          string
+	NumVals           int
+	WithdrawalAddrs   []string
+	FeeRecipientAddrs []string
+}
+
+func newAddValidatorsCmd(runFunc func(context.Context, addValidatorsConfig) error) *cobra.Command {
+	var config addValidatorsConfig
+
+	cmd := &cobra.Command{
+		Use:   "add-validators-solo",
+		Short: "Creates and adds new validators to a solo distributed validator cluster",
+		Long:  `Creates and adds new validators to a distributed validator cluster. It generates keys for the new validators and also generates a new cluster state file with the legacy_lock and add_validators mutations. It is executed by a solo operator cluster.`,
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runFunc(cmd.Context(), config)
+		},
+	}
+
+	bindAddValidatorsFlags(cmd, &config)
+
+	return cmd
+}
+
+func bindAddValidatorsFlags(cmd *cobra.Command, config *addValidatorsConfig) {
+	cmd.Flags().IntVar(&config.NumVals, "num-validators", 1, "The count of new distributed validators to add in the cluster.")
+	cmd.Flags().StringVar(&config.Lockfile, "lock-file", ".charon/cluster-lock.json", "The path to the legacy cluster lock file defining distributed validator cluster.")
+	cmd.Flags().StringSliceVar(&config.FeeRecipientAddrs, "fee-recipient-addresses", nil, "Comma separated list of Ethereum addresses of the fee recipient for each new validator. Either provide a single fee recipient address or fee recipient addresses for each validator.")
```

Comment:
validation will error on 1 address if multiple validators...?

Reply:
Yeah, good catch. I modified the code and added tests for this case, ie, `num_validators > 1` and `len(fee-recipient-addrs) == 1`, since the same address will be used for all validators.